spec/unit/cookbook_version_spec.rb

```diff
@@ -77,77 +79,85 @@
   describe "with a cookbook directory named tatft" do
     MD5 = /[0-9a-f]{32}/
 
-    before do
-      @cookbook = Hash.new { |hash, key| hash[key] = [] }
-
-      @cookbook_root = File.join(CHEF_SPEC_DATA, "cb_version_cookbooks", "tatft")
-
-      # Dunno if the paths here are representitive of what is set by CookbookLoader...
-      @cookbook[:attribute_filenames]   = Dir[File.join(@cookbook_root, "attributes", "**", "*.rb")]
-      @cookbook[:definition_filenames]  = Dir[File.join(@cookbook_root, "definitions", "**", "*.rb")]
-      @cookbook[:file_filenames]        = Dir[File.join(@cookbook_root, "files", "**", "*.tgz")]
-      @cookbook[:recipe_filenames]      = Dir[File.join(@cookbook_root, "recipes", "**", "*.rb")]
-      @cookbook[:template_filenames]    = Dir[File.join(@cookbook_root, "templates", "**", "*.erb")]
-      @cookbook[:library_filenames]     = Dir[File.join(@cookbook_root, "libraries", "**", "*.rb")]
-      @cookbook[:resource_filenames]    = Dir[File.join(@cookbook_root, "resources", "**", "*.rb")]
-      @cookbook[:provider_filenames]    = Dir[File.join(@cookbook_root, "providers", "**", "*.rb")]
-      @cookbook[:root_filenames]        = Array(File.join(@cookbook_root, "README.rdoc"))
-      @cookbook[:metadata_filenames]    = Array(File.join(@cookbook_root, "metadata.json"))
-    end
+    let(:cookbook_paths_by_type) do
+      {
+        # Dunno if the paths here are representitive of what is set by CookbookLoader...
+        all_unignored_files:  Dir[File.join(cookbook_root, "**", "*.rb")],
```

Comment:
Yeah, I'm a bit more than 51% -- not super wild about it, but it does feel like its the thing we decided to name in that RFC for our $REASONS, so we should do that until we discover its unworkably awkward...